src/tiled/newtemplatedialog.cpp

```diff
@@ -20,20 +20,36 @@
  */
 
 #include "newtemplatedialog.h"
+#include "objecttemplatemodel.h"
+#include "preferences.h"
+#include "templatemanager.h"
+#include "tmxmapformat.h"
 #include "ui_newtemplatedialog.h"
 
+#include <QFileDialog>
+#include <QMessageBox>
 #include <QPushButton>
 
 using namespace Tiled;
 using namespace Tiled::Internal;
 
-NewTemplateDialog::NewTemplateDialog(QList<QString> groupNames, QWidget *parent) :
+NewTemplateDialog::NewTemplateDialog(const QList<QString> &groupNames, const QString &objectName, QWidget *parent) :
     QDialog(parent),
     mUi(new Ui::NewTemplateDialog)
 {
     mUi->setupUi(this);
+
+    mObjectName = objectName;
+    mUi->templateNameCheckBox->setEnabled(!objectName.isEmpty());
+    connect(mUi->templateNameCheckBox, &QCheckBox::toggled,
+            this, &NewTemplateDialog::updateTemplateName);
+
+    connect(mUi->createGroupButton, &QPushButton::pressed,
+            this, &NewTemplateDialog::newTemplateGroup);
+
     mUi->groupsComboBox->addItems(groupNames);
```

Comment:
Note that you can just do this:
```c++
    auto model = ObjectTemplateModel::instance();
    mUi->groupsComboBox->setModel(model);
```
This way, there is also no need to update the list of items after adding a new template group.